packages/legend-application-studio/src/components/editor/panel-group/DevToolPanel.tsx

```diff
@@ -81,6 +81,27 @@ export const DevToolPanel = observer(() => {
             !isValidUrl(engineConfig.baseUrl ?? '') ? 'Invalid URL' : ''
           }
         />
+        {Boolean(
+          editorStore.applicationStore.config.options
+            .TEMPORARY__serviceRegistrationConfig.length,
+        ) && (
+          <PanelFormTextField
+            name="Engine client service registration base URL"
+            value={engineConfig.baseUrlForServiceRegistration ?? ''}
+            isReadOnly={false}
+            update={(value: string | undefined): void =>
+              engineConfig.setBaseUrlForServiceRegistration(
+                value === '' ? undefined : value,
+              )
+            }
+            errorMessage={
```

Comment:
confused where exactly we are using `baseUrlForServiceRegistration` during the actual registration of the service.